.github/libs/versionUpdater.js

```diff
@@ -0,0 +1,98 @@
+const semanticVersionLevels = {
+    major: 'MAJOR',
+    minor: 'MINOR',
+    patch: 'PATCH',
+    build: 'BUILD',
+};
+const maxIncrements = 999;
+
+/**
+ * Transforms a versions string into a number
+ *
+ * @param {String} versionString
+ * @returns {Array}
+ */
+const getVersionNumberFromString = (versionString) => {
+    const [version, build] = versionString.split('-');
+    const [major, minor, patch] = version.split('.').map(n => Number(n));
+    return [major, minor, patch, build ? Number(build) : undefined];
+};
+
+/**
+ * Transforms version numbers components into a version string
+ *
+ * @param {Number} major
+ * @param {Number} minor
+ * @param {Number} patch
+ * @param {Number} build
```

Comment:
```suggestion
 * @param {Number} [build]
```

Reply:
Why the brackets?

Reply:
In a JSDoc these brackets mark a parameter as optional. NAB

Reply:
![image](https://user-images.githubusercontent.com/47436092/110187465-4a1b4180-7dcd-11eb-8fa7-f2a305dfd781.png)

Reply:
Ah, sorry. Like I said before I usually only work with Typescript and these things are done in a different way.